### PR TITLE
ROX-22101: Remove gogo protobuf imports from central NetworkGraph

### PR DESCRIPTION
**Pull request by rhybrillou**
## Description

The goal is to remove direct calls to the gogo protobuf library and have all of them happen through an abstraction layer in order to ease the migration to a new protobuf library.

The current PR aims at changing the functions around network graph management in central to use time rather than gogo timestamps as parameters or return values.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

TODO(replace-me)  
Use this space to explain **how you validated** that **your change functions exactly how you expect it**.
Feel free to attach JSON snippets, curl commands, screenshots, etc. Apply a simple benchmark: would the information you
provided convince any reviewer or any external reader that you did enough to validate your change.

It is acceptable to assume trust and keep this section light, e.g. as a bullet-point list.

It is acceptable to skip testing in cases when CI is sufficient, or it's a markdown or code comment change only.  
It is also acceptable to skip testing for changes that are too taxing to test before merging. In such case you are
responsible for the change after it gets merged which includes reverting, fixing, etc. Make sure you validate the change
ASAP after it gets merged or explain in PR when the validation will be performed.  
Explain here why you skipped testing in case you did so.

Have you created automated tests for your change? Explain here which validation activities you did manually and why so.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: central/networkgraph/flow/datastore/internal/store/testcommon/flow.go [hunk 1]
```diff
@@ -43,8 +43,8 @@ func (suite *FlowStoreTestSuite) SetupSuite() {
 func (suite *FlowStoreTestSuite) TestStore() {
 	// Postgres timestamp only goes to the microsecond level, so we need to truncate these test times
 	// to ensure the comparisons of the results works correctly.
-	t1 := time.Now().Add(-5 * time.Minute).Truncate(time.Microsecond)
-	t2 := time.Now().Truncate(time.Microsecond)
+	t1 := time.Now().UTC().Add(-5 * time.Minute).Truncate(time.Microsecond)
+	t2 := time.Now().UTC().Truncate(time.Microsecond)
```
Comment (janisz) [inline central/networkgraph/flow/datastore/internal/store/testcommon/flow.go:47]: Why we need `UTC()` ?

**Review by janisz — APPROVED** [review]

File: central/networkgraph/flow/datastore/internal/store/postgres/store_test.go [hunk 2]
```diff
@@ -95,15 +90,16 @@ func (s *NetworkflowStoreSuite) TestStore() {
...
+	networkFlow.LastSeenTimestamp = protocompat.GetProtoTimestampFromSeconds(2)
 	s.NoError(s.store.UpsertFlows(s.ctx, []*storage.NetworkFlow{networkFlow}, zeroTs))
 	foundNetworkFlows, _, err = s.store.GetAllFlows(s.ctx, nil)
 	s.NoError(err)
 	s.Len(foundNetworkFlows, 1)
 	s.Equal(networkFlow, foundNetworkFlows[0])
 
 	// Check the get all flows by since time
-	foundNetworkFlows, _, err = s.store.GetAllFlows(s.ctx, getTimestamp(3))
+	time3 := time.Unix(3, 0)
```
Comment (vikin91) [inline central/networkgraph/flow/datastore/internal/store/postgres/store_test.go:101]: Why not `protocompat.GetProtoTimestampFromSeconds(3)` as in the other places where we change from `getTimestamp(#)`?

`time3` is here "3 seconds after January 1, 1970 UTC".

File: central/networkgraph/service/service_impl.go [hunk 3]
```diff
@@ -429,7 +429,9 @@ func (s *serviceImpl) addDeploymentFlowsToGraph(
 		}
 	}
 
-	flows, _, err := flowStore.GetMatchingFlows(networkGraphGenElevatedCtx, pred, request.GetSince())
+	since := protocompat.ConvertTimestampToTimeOrNil(request.GetSince())
```
Comment (vikin91) [inline central/networkgraph/service/service_impl.go:432]: (nit) is the introduction of the `since` var necessary? I think, we may remove it.

**Review by vikin91 — COMMENTED** [review]
Looks good! Before I accept, could you please check the non-nit question in the comments?

**Review by vikin91 — APPROVED** [review]
Thanks for explaning!

**Pull request merged**